plugin/db/mysql/mysql.go

```diff
@@ -913,21 +913,88 @@ func (driver *Driver) Dump(ctx context.Context, database string, out io.Writer,
 }
 
 // Restore restores a database.
-func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner) (err error) {
+func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner, config db.RestoreConfig) (err error) {
 	txn, err := driver.db.BeginTx(ctx, nil)
 	if err != nil {
 		return err
 	}
 	defer txn.Rollback()
 
-	f := func(stmt string) error {
+	fnExecuteStmt := func(stmt string) error {
 		if _, err := txn.Exec(stmt); err != nil {
 			return err
 		}
 		return nil
 	}
 
-	if err := util.ApplyMultiStatements(sc, f); err != nil {
+	// restore to ghost tables, need to replace table names
+	if config.IsGhostTable {
+		reInsert := regexp.MustCompile(`INSERT INTO ` +
```

Comment:
This is something we can add some unit tests for.

Btw, this is something that I might care about performance since regex, garbage collection, etc. Since the backup file is generated by us, we know 
1) the table name has no database prefix;
2) we can use string prefix to determine the type of DDL/DML statements such as strings.HasPrefix(stmt, "CREATE TABLE").

Reply:
I'll extract the match and replace logic into a testable function and add some unit tests.

As for the performance issue, the `CREATE TABLE` statement should be small in size, and the regexp is compiled once and used many times. I suppose it's not an issue. And regex makes the matching logic clear.